### PR TITLE
feat(ci): Enable releases through semantic release

### PR DESCRIPTION
**Pull request by esatterwhite**
Include semantic release as a dev dependency to automate package
releases. Updated Jenkinsfile to run release process through semantic
release

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -51,26 +52,34 @@ pipeline {
...
-            usernameVariable: 'TWINE_USERNAME',
-            passwordVariable: 'TWINE_PASSWORD'
+        withCredentials([
+          string(
+            credentialsId: 'pypi-token',
+            variable: 'PYPI_TOKEN',
+          ),
+          string(
+            credentialsId: 'github-api-token',
+            variable: 'GITHUB_TOKEN',
```
Comment (esatterwhite) [inline Jenkinsfile:70]: Is there a better way to do this? `string` feels dangerous

File: Jenkinsfile [hunk 2]
```diff
@@ -51,26 +52,34 @@ pipeline {
...
           )
         ]){
-          sh 'python setup.py sdist'
-          sh 'twine upload dist/logdna-*.tar.gz'
+          sh "echo ${BRANCH_NAME}"
+          sh "echo ${JENKINS_URL}"
```
Comment (esatterwhite) [inline Jenkinsfile:74]: can remove

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+.PHONY:package
+package:                         ## Generate a python sdist and wheel
+	poetry build
+
+.PHONY:release
+release: clean                   ## run semantic release build and publish results to github + pypi based on unreleased commits
+	poetry run semantic-release publish
+
+.PHONY:release-dry
+release-dry: clean changelog         ## run semantic release in noop mode
```
Comment (esatterwhite) [inline Makefile:45]: spacing is off by a couple

File: Jenkinsfile (hunk 1 above)
Comment (gregswift) [reply Jenkinsfile:70]: The  [main way](https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#secret-text) to handle both of these to replace this `withCredentials([]){}` block with:

```
environment {
  GITHUB_TOKEN = credentials('github-api-token')
  PYPI_TOKEN = credentials('pypi-token')
}
```

Comment (esatterwhite) [reply Jenkinsfile:70]: I'm not entirely sure that works with a docker context

https://issues.jenkins.io/browse/JENKINS-42369

File: pyproject.toml [hunk 4]
```diff
@@ -26,6 +34,7 @@ post_test = "python scripts/json_coverage.py"
 lint = "flake8 --doctests"
 "lint:fix" =  "yapf -r -i logdna scripts tests"
 "post_lint:fix" = "task lint"
+release = "semantic-release publish"
```
Comment (mdeltito) [inline pyproject.toml:37]: is this used/useful now that this is driven by the makefile? or rather, should the makefile instead run these tasks vs calling `semantic-release` directly?

Comment (esatterwhite) [reply pyproject.toml:37]: update makefile to run this.

**Review by mdeltito — DISMISSED** [review]
LGTM

File: Jenkinsfile (hunk 1 above)
Comment (esatterwhite) [reply Jenkinsfile]: Nevermind, It looks like everything is picked up, I think this jira is in reference to a slightly different problem

**Review by mdeltito — APPROVED** [review]

**Review by david-ly — DISMISSED** [review]

File: setup.py [hunk 5]
```diff
@@ -7,9 +7,8 @@
 with open(path.join(this_directory, 'README.md'), 'rb') as f:
     long_description = f.read().decode('utf-8')
 
-with open("{p}{s}logdna{s}VERSION"
-          .format(p=this_directory, s=sep)
-          ) as f:
+kwargs = {"dir": this_directory, "sep": sep}
+with open("{dir}{sep}logdna{sep}VERSION".format(**kwargs)) as f:
```
Comment (esatterwhite) [inline setup.py:11]: This could also be `path.join`

**Review by mdeltito — APPROVED** [review]

**Review by david-ly — APPROVED** [review]

**Pull request merged**